packages/contracts/src/dollar/facets/DirectGovernanceFarmerFacet.sol

```diff
@@ -43,22 +43,33 @@ contract DirectGovernanceFarmerFacet is Modifiers {
         );
     }
 
-    /*function onERC1155Received(
+    /**
+     * @notice Handles the receipt of a single ERC1155 token type. This function is
+     * called at the end of a `safeTransferFrom` after the balance has been updated.
+     *
+     * NOTE: To accept the transfer, this must return
+     * `bytes4(keccak256("onERC1155ReceivedDirectGovernance(address,address,uint256,uint256,bytes)"))`
+     * (i.e. 0x7abeba72, or its own function selector).
+     *
+     * @return `bytes4(keccak256("onERC1155ReceivedDirectGovernance(address,address,uint256,uint256,bytes)"))` if transfer is allowed
+     */
+
+    function onERC1155ReceivedDirectGovernance(
         address operator,
         address from,
         uint256 id,
         uint256 value,
         bytes calldata data
-    ) external view returns (bytes4) {
+    ) external pure returns (bytes4) {
         return
-            LibDirectGovernanceFarmer.onERC1155Received(
+            LibDirectGovernanceFarmer.onERC1155ReceivedDirectGovernance(
```

Comment:
@rndquu renamed cause libcreditnft has one and diamond it's detecting naming clash

Reply:
The whole purpose of the `onERC1155Received()` method is to make sure that the contract supports (i.e. can receive) ERC1155 tokens so renaming the `onERC1155Received()` to `onERC1155ReceivedDirectGovernance()`:
1. Violates [IERC1155Receiver](https://docs.openzeppelin.com/contracts/3.x/api/token/erc1155#IERC1155Receiver)
2. Breaks the contract in terms that it wouldn't support accepting ERC1155 tokens

The thing is that [DirectGovernanceFarmerFacet](https://github.com/ubiquity/ubiquity-dollar/blob/c41a33daf263f569f9c7ef8a6ba3d7ff320ce7ed/packages/contracts/src/dollar/facets/DirectGovernanceFarmerFacet.sol) operates only with ERC20 tokens so I don't really understand why we even need to conform to `IERC1155Receiver`. I guess we can simply delete the [onERC1155ReceivedDirectGovernance](https://github.com/ubiquity/ubiquity-dollar/blob/c41a33daf263f569f9c7ef8a6ba3d7ff320ce7ed/packages/contracts/src/dollar/facets/DirectGovernanceFarmerFacet.sol#L57) method.

Reply:
@rndquu

Highlighting some things:

1. I think the main idea to have this method (as it is implemented in actual contract) is because and ERC1155 transfer happens under the hood when the user receives [LP Pool](https://github.com/ubiquity/ubiquity-dollar/blob/904396c1596128bb8f4b3bffde0ccb1f555a8451/packages/contracts/src/dollar/DirectGovernanceFarmer.sol#L184) , at staking share transfer LibCreditNFTManager also [implements](https://github.com/ubiquity/ubiquity-dollar/blob/904396c1596128bb8f4b3bffde0ccb1f555a8451/packages/contracts/src/dollar/libraries/LibCreditNftManager.sol#L242), the names cannot be the same in diamond facet
2. I couldn't just simple remove up until this PR happen due to keeping the originality of the contract
3. Removing/purging this interface is an idea coming on this PR

Reply:
> I think the main idea to have this method (as it is implemented in actual contract) is because and ERC1155 transfer happens under the hood when the user receives [LP Pool](https://github.com/ubiquity/ubiquity-dollar/blob/904396c1596128bb8f4b3bffde0ccb1f555a8451/packages/contracts/src/dollar/DirectGovernanceFarmer.sol#L184) , at staking share transfer LibCreditNFTManager also [implements](https://github.com/ubiquity/ubiquity-dollar/blob/904396c1596128bb8f4b3bffde0ccb1f555a8451/packages/contracts/src/dollar/libraries/LibCreditNftManager.sol#L242), the names cannot be the same in diamond facet

I can't find where any ERC1155 transfers happen [here](https://github.com/ubiquity/ubiquity-dollar/blob/904396c1596128bb8f4b3bffde0ccb1f555a8451/packages/contracts/src/dollar/DirectGovernanceFarmer.sol#L184) under the hood. Could you show the related code?

> I couldn't just simple remove up until this PR happen due to keeping the originality of the contract

Again, it doesn't make sense in renaming `onERC1155Received()` to any other value because in this case the contract (the facet in our case) does not conform to [IERC1155Receiver](https://docs.openzeppelin.com/contracts/3.x/api/token/erc1155#IERC1155Receiver) hence `onERC1155ReceivedDirectGovernance()` **will never be called**.

I've just removed `onERC1155ReceivedDirectGovernance()` along with the `onERC1155Received()` [test](https://github.com/ubiquity/ubiquity-dollar/blob/c41a33daf263f569f9c7ef8a6ba3d7ff320ce7ed/packages/contracts/test/diamond/facets/DirectGovernanceFarmerFacet.t.sol#L111) and all other tests are passing.

Reply:
Yeah it doesn't make sense either, and then im positive too in getting rid of this interface on this facet

Reply:
done in https://github.com/ubiquity/ubiquity-dollar/pull/841/commits/3554844f8f5dd7491b1d9f93d9b29c68cef4112f

Reply:
i have removed the function but for your info here is a original comment (that i have not edited not update about the nft transfer)
https://github.com/ubiquity/ubiquity-dollar/pull/841/files#diff-ec25ab7a74d12b4768ebfb4b5fe165eaff5dc3b3d90690074dc05a0fcc38b7a1L329